src/esp/assets/ResourceManager.cpp

```diff
@@ -1577,55 +1574,36 @@ gfx::PbrMaterialData::uptr ResourceManager::buildPbrShadedMaterialData(
     }
   }
 
-  // CAREFUL:
-  // certain texture will be stored "multiple times" in the `finalMaterial`;
-  // (e.g., roughnessTexture will be stored in noneRoughnessMetallicTexture,
-  // roughnessTexture, metallicTexture)
-  // It is OK! No worries! Such duplication (or "conflicts") will be handled in
-  // the PbrShader (see PbrMaterialData for more details)
-
   // roughness
   if (material.hasAttribute(Mn::Trade::MaterialAttribute::Roughness)) {
     finalMaterial->roughness = material.roughness();
   }
   if (material.hasRoughnessTexture()) {
     finalMaterial->roughnessTexture =
         textures_.at(textureBaseIndex + material.roughnessTexture()).get();
-    if (!material.hasAttribute(Mn::Trade::MaterialAttribute::Roughness)) {
-      finalMaterial->roughness = 1.0f;
-    }
   }
 
-  // metalness
+  // metallic
   if (material.hasAttribute(Mn::Trade::MaterialAttribute::Metalness)) {
     finalMaterial->metallic = material.metalness();
   }
   if (material.hasMetalnessTexture()) {
     finalMaterial->metallicTexture =
         textures_.at(textureBaseIndex + material.metalnessTexture()).get();
-    if (!material.hasAttribute(Mn::Trade::MaterialAttribute::Metalness)) {
-      finalMaterial->metallic = 1.0f;
-    }
   }
 
-  // packed textures
-  if (material.hasOcclusionRoughnessMetallicTexture()) {
-    // occlusionTexture, roughnessTexture, metalnessTexture are pointing to the
-    // same texture ID, so just occlusionTexture
-    finalMaterial->occlusionRoughnessMetallicTexture =
-        textures_.at(textureBaseIndex + material.occlusionTexture()).get();
-  }
-
-  if (material.hasNoneRoughnessMetallicTexture()) {
-    // roughnessTexture, metalnessTexture are pointing to the
-    // same texture ID, so just roughnessTexture
-    finalMaterial->noneRoughnessMetallicTexture =
-        textures_.at(textureBaseIndex + material.roughnessTexture()).get();
+  // sanity check when both metallic and roughness materials are presented
+  if (material.hasMetalnessTexture() && material.hasRoughnessTexture()) {
+    CORRADE_ASSERT(
+        finalMaterial->metallicTexture == finalMaterial->roughnessTexture,
```

Comment:
@mosra: I still use: `finalMaterial->metallicTexture == finalMaterial->roughnessTexture` instead of 
`material.hasNoneRoughnessMetallicTexture() | material.hasOcclusionRoughnessMetallicTexture() | material.hasNormalRoughnessMetallicTexture()` for code simplicity.

Reply:
Code simplicity? No, the main issue with comparing just IDs is that it's *not enough*. It doesn't check that:

- both use the same texture coordinate attribute,
- both have the same texture transformation, and
- the metalness is in B and roughness is in G

While the 1st and 3rd case are rather unlikely with models that are currently circulating around the web, the second case sometimes happens, especially with models that were processed with meshoptimizer / gltfpack. So *please* can you use `hasNoneRoughnessMetallicTexture()` to have *everything* checked? Thanks.

Reply:
I see. So it means in the `hasNoneRoughnessMetallicTexture`, it actually does more sanity checks than it literally suggests. I do not know this before.

Question: other than `hasNoneRoughnessMetallicTexture`, do I need to also check:

`material.hasNoneRoughnessMetallicTexture() | material.hasOcclusionRoughnessMetallicTexture() | material.hasNormalRoughnessMetallicTexture()`?